api/handlers.go

```diff
@@ -53,3 +56,49 @@ func scale(c echo.Context) error {
 	}
 	return c.NoContent(http.StatusCreated)
 }
+
+func updateCertificate(c echo.Context) error {
+	makeInvalidKeyOrCertificateResponse := func(err error) error {
+		logrus.Error(err)
+		return c.String(http.StatusPreconditionFailed, "Invalid key or certificate")
```

Comment:
About return the `err` to the framework directly, I was worried since it could return sensitive information about the server condition for its clients. Make sense? 

About the 412 status code, I was following the old RPaaS web API. I agree this code isn't good. I'll replace that on next commits. Thanks for the explanation!

Reply:
My bad, the Echo web framework does not show the internal error when returning the a non-nil error.